### PR TITLE
pkg/cmsis-nn: add support to RIOT

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR adds support for the neural network API provided by ARM in the [CMSIS repository](https://github.com/ARM-software/CMSIS_5).
A test application is provided for performing the detection of an object in an RGB image from the [CIFAR10 dataset](https://www.cs.toronto.edu/~kriz/cifar.html). This sample application is adapted from the [provided example in the upstream repo](https://github.com/ARM-software/CMSIS_5/tree/develop/CMSIS/NN/Examples/ARM/arm_nn_examples/cifar10).
The image is embedded in the generated firmware using the BLOB mechanism.

Note that the package could have been done based on the existing cmsis-dsp but I wanted to use the latest release 5.6 and the existing cmsis-dsp is using 5.4 and doesn't seem to be that easy to update.
I already have a branch that does that but it's not working on all supported CortextM (like M0).

cc @emmanuelsearch, as we already talked about that before.

### Testing procedure

The following command should succeed:
```
make BOARD=<one of the whitelisted board> -C tests/pkg_cmsis-nn flash test
```

### Issues/PRs references

None

### CODE REVIEW

File: tests/pkg_cmsis-nn/Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+USEPKG += cmsis-nn
+
+BLOBS += input
+
+# Boards that were tested and are known to work
+BOARD_WHITELIST := \
+  b-l475e-iot01a \
+  iotlab-m3 \
+  nucleo-l476rg \
+  stm32f723e-disco
```
Comment (benpicco) [inline tests/pkg_cmsis-nn/Makefile]: Just wondering - what's special about these?

Comment (aabadie) [reply tests/pkg_cmsis-nn/Makefile]: I had them with me when I wrote this PR, so I'm sure they work :)

Comment (benpicco) [reply tests/pkg_cmsis-nn/Makefile]: Sure, but are there any requirements to the MCU for this to work or is it just software?

Comment (aabadie) [reply tests/pkg_cmsis-nn/Makefile]: Yes, only with M3 (but this is not explicitly mentioned), M4 and M7. I remember now: that required too much changes in the build system and CPUs to be made more automatic. Something like adding `FEATURES_PROVIDED += arch_cortexm4` (that specific level doesn't exist, yet), etc in CPUs and use `FEATURES_REQUIRED += arch_cortexm` with appropriate `FEATURES_BLACKLIST += arch_cortexm0%`, etc in the package Makefile.dep.

#13174 and #13236 could help but there's work.

Comment (benpicco) [reply tests/pkg_cmsis-nn/Makefile]: Better add this to the comment then.
My `same54-xpro` also recognizes a cat, but it's probably a bad idea to start a list of every CortexM(≥3) board here ;)

Comment (aabadie) [reply tests/pkg_cmsis-nn/Makefile]: Comment added and also added a couple of new boards (same54-xpro, nrf52dk and nrf52832-mdk) that were tested with success.

File: pkg/cmsis-nn/Makefile.include [hunk 2]
```diff
@@ -0,0 +1,6 @@
+INCLUDES += -I$(PKGDIRBASE)/cmsis-nn/CMSIS/NN/Include
+
+# Required for some basic math functions
+INCLUDES += -I$(PKGDIRBASE)/cmsis-nn/CMSIS/DSP/Include
+
+CFLAGS += -Wno-sign-compare
```
Comment (benpicco) [inline pkg/cmsis-nn/Makefile.include]: already in `Makefile`

Comment (aabadie) [reply pkg/cmsis-nn/Makefile.include]: I kept this one and removed the other: the warning occurs in a header file and is not silenced from the package `Makefile`. So I have to silent it globally unfortunately.

**Comment by benpicco** [comment]
Looks good to me - please squash.

**Comment by aabadie** [comment]
Squashed!

**Comment by aabadie** [comment]
Thanks for reviewing and merging @benpicco !

**Pull request merged**